server/py/services/api/tests/unit/db/test_runs.py

```diff
@@ -425,6 +425,46 @@ def test_update_run_success(self):
             assert run["status"]["end_time"] is not None
             assert update_labels_mock.call_count == 0
 
+    def test_store_and_update_run_from_terminal_state_to_non_terminal_state(self):
+        project, name, uid, iteration, run = self._create_new_run(
+            state=mlrun.common.runtimes.constants.RunStates.completed
+        )
+        run = self._db.read_run(self._db_session, uid, project, iteration)
+
+        # Store completed expected to fill end time
+        initial_end_time = run["status"]["end_time"]
+        assert initial_end_time is not None
+
+        self._create_new_run(state=mlrun.common.runtimes.constants.RunStates.running)
```

Comment:
From reading the rest of the test, I'm guessing this line isn't creating a new run but updating the same run, as you are querying the same uid and iteration of the old one and expecting the state to be running. Or am I missing something? If that is indeed the case, maybe use update here instead of create? Or add a comment explaining what's happening.

Reply:
deliberately using create to test the store flow as well. added a comment